python/cudf/cudf/core/dataframe.py

```diff
@@ -3720,6 +3720,98 @@ def sort_values(
+        
+        Returns 
+        -------
+        scalar : when Series.agg is called with single aggregation
+        Series : when DataFrame.agg is called with a single aggrgation
+        DataFrame : when DataFrame.agg is called with several aggregations
+
+        Notes
+        -----
+        Difference from pandas:
+          * Not supporting: *args, **kwargs
+
+        """
+        if axis is not None:
+            raise NotImplementedError("axis not implemented yet")
+
+        dtypes = [self[col].dtype for col in self._column_names]
+        common_dtype = cudf.utils.dtypes.find_common_type(dtypes)
+        df_normalized = self.astype(common_dtype)
+
+        if isinstance(aggs, list):
+            result = cudf.DataFrame()
+            # TODO : Allow simultaneous pass for multi-aggregation as a future optimization
+            for agg in aggs:
+                result[agg] = getattr(df_normalized, agg)()
+
+        elif isinstance(aggs, str):
+            result = cudf.DataFrame()
+            result[aggs] = getattr(df_normalized,aggs)()
+            result = result.T.loc[aggs]
+            result.name = None
+            return result
+
+        elif isinstance(aggs, dict):
+            cols=aggs.keys()
+            if any([callable(val) for val in aggs.values()]):
+                raise NotImplementedError("callable parameter is not implemented yet")
+            elif any([isinstance(val, list) for val in aggs.values()]):
+                idxs=set()
+                for val in aggs.values():
+                    if isinstance(val, list):
+                        idxs.update(val)
+                    elif isinstance(val, str):
+                        idxs.add(val)    
+                idxs = sorted(list(idxs))
+                for agg in idxs:
+                    if agg is callable:
+                        raise NotImplementedError("callable parameter is not implemented yet")
+                result = cudf.DataFrame(index=idxs, columns=cols)
+                for key in aggs.keys():
+                    col = df_normalized[key]
+                    col_empty =  column_empty(len(idxs), dtype=col.dtype, masked=True)
+                    ans = cudf.Series(data=col_empty, index=idxs)
+                    if isinstance(aggs.get(key),list):
+                        # TODO : Allow simultaneous pass for multi-aggregation as a future optimization
+                        for agg in aggs.get(key):
+                            ans[agg] = getattr(col, agg)()
+                    elif isinstance(aggs.get(key),str):
+                        ans[aggs.get(key)] = getattr(col, agg)()
+                    result[key] = ans
+            elif all([isinstance(val, str) for val in aggs.values()]):
+                result = cudf.Series(index=cols)
+                for key, value in aggs.items():
+                    col = df_normalized[key]
+                    result[key]= getattr(col, value)()
```

Comment:
For the fall through branch of this `if-elif-else`, let's also raise a `ValueError('values of dict must be a string or list')`.